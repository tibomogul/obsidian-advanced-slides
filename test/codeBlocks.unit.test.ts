import { ObsidianMarkdownPreprocessor } from 'src/obsidianMarkdownPreprocessor';
import { obsidianUtils as utilsInstance } from './__mocks__/mockObsidianUtils';
import { prepare } from './testUtils';

test('Code Block Syntax > Headers', () => {
	const input = `
\`\`\`
USER $USER_NAME:$USER_NAME
\`\`\`

The above does not show backticks
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Code Block Syntax > Headers', () => {
	const input = ` \`\`\`
USER $USER_NAME:$USER_NAME
\`\`\`

The above does show backticks
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});

test('Code Block Syntax > Headers', () => {
	const input = `\`\`\`
USER $USER_NAME:$USER_NAME
\`\`\`

The above does not show backticks
`;

	const { options, markdown } = prepare(input);
	const sut = new ObsidianMarkdownPreprocessor(utilsInstance);

	return expect(sut.process(markdown, options)).toMatchSnapshot();
});