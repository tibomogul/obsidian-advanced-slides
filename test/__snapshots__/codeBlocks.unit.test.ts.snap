// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Block Syntax > Headers 1`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

\`\`\`
USER $USER_NAME:$USER_NAME
\`\`\`

The above does not show backticks
</div>"
`;

exports[`Code Block Syntax > Headers 2`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

\`\`\`
USER \`$USER_NAME:$\`USER_NAME
\`\`\`

The above does show backticks
</div>"
`;

exports[`Code Block Syntax > Headers 3`] = `
"<!-- .slide: class=\\"drop\\" -->
<div class=\\"\\" style=\\"position: absolute; left: 0px; top: 0px; height: 700px; width: 960px; min-height: 700px; display: flex; flex-direction: column; align-items: center; justify-content: center\\" absolute=\\"true\\">

\`\`\`
USER $USER_NAME:$USER_NAME
\`\`\`

The above does not show backticks
</div>"
`;
